signing-with-metamask/index.md

```diff
@@ -0,0 +1,89 @@
+
+> But first, we will understand a digital signature and data signing in web3.
+
+A digital signature is a mathematical mechanism used to verify the authenticity and integrity of digital data. Signing data in web3 refers to using digital signatures to authenticate and authorize transactions on the blockchain. It combines a user's private key, which is kept secret, with the data to be signed, and then encrypts it using a one-way hash function. Anyone with access to the user's public key and the original data can verify the resulting signature.
+
+Digital signatures are used to sign transactions in web3 before they are broadcasted to the blockchain network. This signing ensures that only the authorized user can initiate a transaction and that the transaction has not been tampered with in transit. 
+
+Signing data is also used in smart contracts to prove that a user has the necessary permissions to execute a particular function or transfer a specific asset. Digital signatures are a fundamental aspect of web3 and are crucial for maintaining the security and integrity of the blockchain network.
+
+> And what does MetaMask have to do with all of this?
+
+MetaMask is a popular browser extension that allows users to access the Ethereum blockchain and interact with decentralized applications (dApps). Signing data with MetaMask is a feature that allows users to sign digital data with their Ethereum private key, providing a way to prove ownership of an Ethereum address. This ownership means that they effectively have proven to the network that they have the authority to transfer the specified assets.
+
+When you sign data with MetaMask, your private key never leaves your device, so your key is safe from hackers and other malicious actors who might try to steal it. MetaMask uses a technique known as "hashing" to ensure that your data cannot be tampered with and is secure.
+
+Below are some of the critical milestones in the journey of signing data and transactions with MetaMask:
+
+2016: MetaMask was first released as a browser extension for Google Chrome, allowing users to interact with the Ethereum blockchain and sign transactions with their private keys.
+2017: MetaMask added support for signing data and messages, allowing users to prove ownership of an Ethereum address.
+2018: MetaMask introduced the ability to sign typed data, a specific format for structured data specified by the EIP-712 standard. This feature enabled users to sign more structured and organized data.
+2019: MetaMask introduced the ability to sign and send transactions directly from the browser extension, eliminating the need for users to copy and paste the transaction data into a separate signing tool.
+2020: MetaMask introduced the ability to sign transactions using hardware wallets, such as Trezor and Ledger, providing users with an additional layer of security.
```

Comment:
Ledger support was added in 2018: https://github.com/MetaMask/metamask-extension/blob/develop/CHANGELOG.md#4100---2018-09-18
As was Trezor support: https://github.com/MetaMask/metamask-extension/blob/develop/CHANGELOG.md#490---2018-08-07